libnext/src/models/nodes.ts

```diff
@@ -0,0 +1,72 @@
+import { HttpMethod } from "./http";
+import { DataType } from "./types";
+
+export interface ContractNode {
+  api: ApiNode;
+  endpoints: EndpointNode[];
+  types: TypeNode[];
+}
+
+export interface TypeNode {
+  name: string;
+  description?: string;
+  type: DataType;
+}
+
+export interface ApiNode {
+  name: string;
+  description?: string;
+}
+
+export interface EndpointNode {
+  name: string;
+  description?: string;
+  method: HttpMethod;
+  path: string;
+  request?: RequestNode;
+  responses: ResponseNode[];
+  defaultResponse?: DefaultResponseNode;
+}
+
+export interface RequestNode {
+  headers: HeaderNode[];
+  pathParams: PathParamNode[];
+  queryParams: QueryParamNode[];
+  body?: BodyNode;
+}
+
+export interface ResponseNode extends DefaultResponseNode {
+  status: number;
```

Comment:
Why not `status?` in `DefaultResponseNode` and just have one type?

Reply:
Whilst it may be a bit trivial, it does make the internal representation a bit more typesafe

Reply:
I think I understand why I was confused. `ResponseNode` is not semantically extending `DefaultResponseNode`: they're separate concepts. You're using `extends` as a convenient way to reuse code. I wonder if instead the code should be duplicated?

Either way, I think it's worth adding a bit of JSDoc to explain what each of these two types means.